public/app/plugins/panel/canvas/editor/element/utils.ts

```diff
@@ -77,3 +82,9 @@ const getData = (api: APIEditorConfig) => {
 
   return data;
 };
+
+const requestMatchesGrafanaOrigin = (requestEndpoint: string) => {
+  const requestURL = new URL(requestEndpoint);
+  const grafanaURL = new URL(window.location.href);
```

Comment:
Is there a way for window.location.href to be spoofed / hard coded to something else in the request? 🤔

Reply:
Would probably be safer to get the origin from the context.

Reply:
After looking into it more, window.location should be fine to use. [It is read-only](https://developer.mozilla.org/en-US/docs/Web/API/Window/location) and modifying it in the client will result in a redirect.

Reply:
tested this locally and confirmed this behavior 👍 it seems for case where we are reading the value directly in logic it should be safe

@dannykopping is window.location.href safe to use as a source of truth for detecting url?